serverlessworkflow/modules/ROOT/pages/integrations/expose-metrics-to-prometheus.adoc

```diff
@@ -1 +1,123 @@
-//= Exposing the workflow base metrics to Prometheus
\ No newline at end of file
+Exposing the workflow base metrics to Prometheus on Openshift
+=============================================================
+11-03-2022
+:compat-mode!:
+// Metadata:
+:description: Exposing the workflow base metrics to Prometheus
+:keywords: kogito, workflow, quarkus, metrics, prometheus
+// links
+:openshift_micrometer_url: https://quarkus.io/blog/micrometer-prometheus-openshift
+:dashbuilder_url: https://www.dashbuilder.org/
+:grafana_url: https://grafana.com/
+:quarkus_micrometer_url: https://quarkus.io/guides/micrometer
+:openshift_monitoring_url: https://docs.openshift.com/container-platform/4.7/monitoring/enabling-monitoring-for-user-defined-projects.html
+:prometheus_operator_url: https://prometheus-operator.dev/
+
+Kogito Serverless Workflow generate metrics that can be consumed by Prometheus and visualized by dashboard tools such as link:{openshift_micrometer_url}[Openshift], link:{dashbuilder_url}[Dashbuilder] and link:{grafana_url}[Grafana].
+
+This document describes how to enable and expose metrics to Prometheus.
+
+== Enabling metrics
+
+For adding metrics to your Serverless Workflow application just add the dependency `org.kie.kogito:kogito-addons-quarkus-monitoring-prometheus` and the metrics will be available in `/q/metrics` URL.
+
+.Dependency to be added to `pom.xml` to enable metrics
+[source,xml]
+----
+<dependency>
+    <groupId>org.kie.kogito</groupId>
+    <artifactId>kogito-addons-quarkus-monitoring-prometheus</artifactId>
+</dependency>
+----
+
+== Consuming metrics from Openshift
+
+If your Serverless Workflow server is running on Openshift you can use it to monitor your application.
+
+To consume metrics from Openshift you must first link:{openshift_monitoring_url}[enable monitoring for user-defined projects]. Then you need to create a service monitor. Here's a sample configuration:
+
+
+.service-monitor.yaml:
+[source,yaml]
+----
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+    k8s-app: prometheus-app-monitor
+  name: prometheus-app-monitor
+  namespace: my-project
+spec:
+  endpoints:
+  - interval: 30s
+    targetPort: 8080
+    path: /q/metrics
+    scheme: http
+  selector:
+    matchLabels:
+      app-with-metrics: 'serverless-workflow-app'
+----
+
+Then apply it:
+
+[source,shell]
+----
+oc apply -f service-monitor.yaml
+----
+
+What we have just done is to create a Service Monitor named `prometheus-app-monitor` that will select apps with the label selector `app-with-metrics: serverless-workflow-app`, make sure that your Serverless Workflow application has this label. Then, OpenShift will call the endpoint `/q/metrics` for all the services labeled with  `app-with-metrics: serverless-workflow-app` every 30 seconds. For more details check the link:{openshift_micrometer_url}:[Monitoring Quarkus apps using Micrometer and Prometheus into OpenShift] guide.
```

Comment:
```
make sure that your Serverless Workflow application has this label
```
How do I do that ?

Reply:
```
Then, OpenShift will call the endpoint `/q/metrics` for all the services labeled with `app-with-metrics: serverless-workflow-app` every 30 seconds
```
Is that really Openshift doing such a thing ? wouldn't that be the Prometheus operator ?